client/src/main/java/io/pravega/client/state/SynchronizerConfig.java

```diff
@@ -27,7 +36,8 @@
 @Builder
 public class SynchronizerConfig implements Serializable {
 
-    private static final long serialVersionUID = 2L;
+    private static final long serialVersionUID = 3L;
```

Comment:
This isn't quite enough to get Java serialization to use our serialization. We have to override it's methods, by implementing a `writeReplace()` and a `readResolve()` method on a "SerializedFrom". Take a look at how this is done at the bottom of `StreamCutImpl` (Which also does it with byte arrays so you can basically copy and paste that code). Or `SegmentImpl` or `StreamImpl` (which do the same thing but with strings)